app/src/androidTest/java/com/jjswigut/matters/data/source/local/MatterDataBaseDaoTest.kt

```diff
@@ -0,0 +1,75 @@
+package com.jjswigut.matters.data.source.local
+
+import androidx.arch.core.executor.testing.InstantTaskExecutorRule
+import androidx.room.Room
+import androidx.test.core.app.ApplicationProvider.getApplicationContext
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.filters.SmallTest
+import com.jjswigut.matters.database.Matter
+import com.jjswigut.matters.database.MatterDatabase
+import com.jjswigut.matters.getOrAwaitValue
+import junit.framework.Assert.assertEquals
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runBlockingTest
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.CoreMatchers.notNullValue
+import org.hamcrest.MatcherAssert.assertThat
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@ExperimentalCoroutinesApi
+@RunWith(AndroidJUnit4::class)
+@SmallTest
+class MatterDataBaseDaoTest {
+
+    // Executes each task synchronously using Architecture Components.
+    @get:Rule
+    var instantExecutorRule = InstantTaskExecutorRule()
+
+
+    private lateinit var database: MatterDatabase
+
+    @Before
+    fun initDb() {
+        // Using an in-memory database so that the information stored here disappears when the
+        // process is killed.
+        database = Room.inMemoryDatabaseBuilder(
+            getApplicationContext(),
+            MatterDatabase::class.java
+        ).allowMainThreadQueries().build()
+    }
+
+    @After
+    fun closeDb() = database.close()
+
+    @Test
+    fun insertMatterObserveLiveData() = runBlockingTest {
+        // GIVEN - insert a task
+        val matter = Matter("title", "description")
+        database.matterDataBaseDao.insert(matter)
+
+        // WHEN - Get the task by id from the database
+        val matterList = database.matterDataBaseDao.getAllMatters()
+
+        assertEquals(matterList.getOrAwaitValue(), matter)
+    }
+
+    @Test
+    fun updateMatterCheckUpdate() = runBlockingTest {
+
+        val originalMatter = Matter("Title", "Content")
+        database.matterDataBaseDao.insert(originalMatter)
+
+        val updatedMatter = Matter("new title", "new content", originalMatter.matterId)
+        database.matterDataBaseDao.update(updatedMatter)
+
+        assertThat(updatedMatter, notNullValue())
```

Comment:
this set of assertions i don't think is testing what you want to test

you created 2 matter objects, original and updated

you insert the first one, and then make an update call to the 2nd one, and then are just asserting that the updated one has the fields you set

what i think you'd want to do is just create one matter object

insert it into the DB

change the properties like title / content on that original matter object

make the update call

get value from the DB

assert that only 1 item exists in DB and it has the updated values